src/scripts/validation.js

```diff
@@ -0,0 +1,155 @@
+    showInputError(
+      formElement,
+      inputElement,
+      inputElement.validationMessage,
+      config
+    );
+  } else {
+    hideInputError(formElement, inputElement, config);
+  }
+}
+
+export function debounce(func, delay = 300) {
+  let timeout;
+  return (...args) => {
+    clearTimeout(timeout);
+    timeout = setTimeout(() => func(...args), delay);
+  };
+}
+
+export function setEventListeners(formElement, config) {
+  if (!formElement) {
+    console.error(
+      "Error: formElement is undefined, cannot set event listeners."
+    );
+    return;
+  }
+  const inputList = Array.from(
+    formElement.querySelectorAll(config.inputSelector)
+  );
+  const buttonElement = formElement.querySelector(config.submitButtonSelector);
+
+  inputList.forEach((inputElement) => {
+    inputElement.addEventListener(
+      "input",
+      debounce(() => {
+        checkInputValidity(formElement, inputElement, config);
+        toggleButtonState(inputList, buttonElement, config);
+      })
+    );
+  });
+}
+
+export function toggleButtonState(inputList, buttonElement, config) {
+  const isFormInvalid = inputList.some((input) => !input.validity.valid);
+  if (!buttonElement) {
+    console.error(
+      "Error: buttonElement is undefined, cannot toggle button state."
+    );
+    return;
+  }
+  buttonElement.classList.toggle(config.inactiveButtonClass, isFormInvalid);
+  buttonElement.disabled = isFormInvalid;
+}
+
+export function resetForm(formElement, config) {
+  if (!formElement) {
+    console.error("Error: resetForm called with undefined formElement.");
+    return;
+  }
+  formElement.reset();
+  const inputList = Array.from(
+    formElement.querySelectorAll(config.inputSelector)
+  );
+  const buttonElement = formElement.querySelector(config.submitButtonSelector);
+
+  inputList.forEach((inputElement) =>
+    hideInputError(formElement, inputElement, config)
+  );
+  if (buttonElement) {
+    buttonElement.classList.add(config.inactiveButtonClass);
+    buttonElement.disabled = true;
+  } else {
+    console.error("Error: buttonElement not found during form reset.");
+  }
+}
+
+export function displayErrorsOnSubmit(formElement, config) {
+  if (!formElement) {
+    console.error(
+      "Error: displayErrorsOnSubmit called with undefined formElement."
+    );
+    return;
+  }
+  const inputList = Array.from(
+    formElement.querySelectorAll(config.inputSelector)
+  );
+  inputList.forEach((inputElement) =>
+    checkInputValidity(formElement, inputElement, config)
+  );
+}
+
+export function enableValidation(config) {
+  const formList = Array.from(document.querySelectorAll(config.formSelector));
+  formList.forEach((formElement) => {
+    setEventListeners(formElement, config);
+    formElement.addEventListener("submit", (event) => {
+      event.preventDefault();
+      displayErrorsOnSubmit(formElement, config);
+      if (formElement.checkValidity()) {
+        console.log("Form submitted successfully!");
```

Comment:
Please, delete `displayErrorsOnSubmit(formElement, config);` because it's not needed in the submit handler. The errors should be shown only during the live validation: when the user is typing anything in the inputs